test/python/dagcircuit/test_dagcircuit.py

```diff
@@ -1224,6 +1224,70 @@ def test_circuit_factors(self):
         self.assertEqual(self.dag.num_tensor_factors(), 2)
 
 
+class TestCircuitControlFlowProperties(QiskitTestCase):
+    """Properties tests of DAGCircuit with control-flow instructions."""
+
+    def setUp(self):
+        super().setUp()
+        qc = QuantumCircuit(5, 1)
+        qc.h(0)
+        qc.measure(0, 0)
+        with qc.if_test((qc.clbits[0], True)) as else_:
+            qc.x(1)
+            qc.cx(2, 3)
+        with else_:
+            qc.x(1)
+            with qc.for_loop(range(3)):
+                qc.z(2)
+                with qc.for_loop((4, 0, 1)):
+                    qc.z(2)
+        with qc.while_loop((qc.clbits[0], True)):
+            qc.h(0)
+            qc.measure(0, 0)
+
+        self.dag = circuit_to_dag(qc)
+
+    def test_circuit_size(self):
+        """Test total number of operations in circuit."""
+        self.assertEqual(self.dag.size(recurse=True), 19)
+        with self.assertRaisesRegex(DAGCircuitError, "Cowardly refusing to give a size"):
```

Comment:
Is there a more informative error message we can provide here? Maybe something like "`DAGCircuit.size` is not unambiguously defined for circuits which contain `ControlFlowOps`. See the `DAGCircuit.size` documentation for more info."

Reply:
Done in 9e2c3d870.